src/Akka.Streams.Kafka/Stages/Consumers/CommitObservationLogic.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Akka.Streams.Kafka.Messages;
+using Akka.Streams.Kafka.Settings;
+using Confluent.Kafka;
+
+namespace Akka.Streams.Kafka.Stages.Consumers;
+
+/// <summary>
+/// Shared commit observation logic for different graph stage logics that facilitate offset commits.
+/// </summary>
+/// <remarks>
+/// Really should be implemented as an interface with default implementations, but we can't have nice
+/// things in .NET Standard 2.0. Therefore this gets injected a dependency instead.
+/// </remarks>
+internal sealed class CommitObservationLogic
```

Comment:
Shared logic for committing - used by the `CommitCollectorStage` and also needs to be incorporated into the transactional consumers too.